tests/libstorage/volumesnapshot.go

```diff
@@ -40,3 +53,41 @@ func NewVolumeSnapshot(name, namespace, sourcePvcName string, snapshotClassName
 		},
 	}
 }
+
+func WaitSnapshotSucceeded(virtClient kubecli.KubevirtClient, namespace string, snapshotName string) *snapshotv1.VirtualMachineSnapshot {
+	var snapshot *snapshotv1.VirtualMachineSnapshot
+	Eventually(func() *snapshotv1.VirtualMachineSnapshotStatus {
+		snapshot, err := virtClient.VirtualMachineSnapshot(namespace).Get(context.Background(), snapshotName, metav1.GetOptions{})
```

Comment:
you added : which as you can see in the original code doesnt exist, this caused the updated of the external snapshot var in line 58 to not be updated so you return an empty snapshot here which causes panic nil ref in the tests.

Reply:
Also the code you moved to this library is not the right place since this file is "volumesnapshot" not "vmsnapshot"

Reply:
fixed thanks a lot!